src/scheduler/check.c

```diff
@@ -1568,12 +1568,17 @@ check_nodes(status *policy, server_info *sinfo, queue_info *qinfo, resource_resv
 		else
 			nodepart = NULL;
 
-		/* if there are nodes assigned to the queue, then check those */
-		if (resresv->is_job && qinfo->has_nodes)
-			ninfo_arr = qinfo->nodes;
-		/* last up we're not in a queue with nodes -- use the unassociated nodes */
-		else
-			ninfo_arr = sinfo->unassoc_nodes;
+		if (resresv->is_job && qinfo->has_partition)
+			ninfo_arr = qinfo->nodes_in_partition;
```

Comment:
Use qinfo->nodes first.  Nodes associated with queues are more important than nodes in the partition.  The server won't allow any nodes to be associated with the queue that aren't in its partition anyway.  Plus make this one long if/else if/else block

Reply:
Implemented.